src/compute-types/src/plan/lowering.rs

```diff
@@ -778,7 +778,7 @@ This is not expected to cause incorrect results, but could indicate a performanc
                     .iter()
                     .any(|(key, _, _)| key == &required_arrangement.0)
                 {
-                    types = Some(input.typ().column_types);
+                    types = Some(types.unwrap_or_else(|| input.typ().column_types));
```

Comment:
I think we could do slightly better, sometimes avoiding the `arity()` call, by doing the following at the top of the function:

```rust
let (plan, keys) = self.lower_mir_expr(input)?;
let arity = types.as_ref().map(|types| types.len()).unwrap_or_else(|| input.arity());
let (threshold_plan, required_arrangement) = ThresholdPlan::create_from(arity);
```